### PR TITLE
[stable/jenkins] Kubernetes 1.9 support as well as automatic apiVersion detection

### PR DESCRIPTION
**Pull request by orkhanM**
#### Which issue this PR fixes

* detect apiVersion automatically and template deployment & rbac manifest correspondingly
* add support for kubernetes versions 1.9 and above for jenkins-master-deployment.yaml

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/jenkins/templates/jenkins-master-deployment.yaml [hunk 1]
```diff
@@ -1,4 +1,8 @@
-apiVersion: extensions/v1beta1
+{{- if semverCompare "^1.9-0" .Capabilities.KubeVersion.GitVersion }}
```
Comment (davidkarlsen) [inline stable/jenkins/templates/jenkins-master-deployment.yaml]: you can use https://github.com/helm/helm/blob/master/docs/chart_template_guide/builtin_objects.md apiVersion.has directly instead

Comment (orkhanM) [reply stable/jenkins/templates/jenkins-master-deployment.yaml]: oooh, good call, I'll test it out and push the update. Thanks for this!

File: stable/jenkins/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: jenkins
 home: https://jenkins.io/
-version: 0.18.1
+version: 0.18.2
```
Comment (davidkarlsen) [inline stable/jenkins/Chart.yaml]: I'd say 0.19.0 since it is a new feature (and people could notice the param is now no longer in use and can remove it).

Comment (orkhanM) [reply stable/jenkins/Chart.yaml]: Ahh, good call. Pushing the change now.

**Comment by orkhanM** [comment]
@davidkarlsen could we get this merged?

**Pull request merged**